pkg/pillar/devicenetwork/arp.go

```diff
@@ -0,0 +1,198 @@
+		return false
+	}
+	return true
+}
+
+func makeArpEntry(outIntf string, macAddr string, addr net.IP, subnet net.IPNet) arpEntry {
+	return arpEntry{
+		outIntf: outIntf,
+		macAddr: macAddr,
+		addr:    addr,
+		subnet:  subnet,
+	}
+}
+
+func arpEntriesDifferent(oldEntries []arpEntry, newEntries []arpEntry) bool {
+	if len(oldEntries) != len(newEntries) {
+		return true
+	}
+
+	for i := 0; i < len(oldEntries); i++ {
+		oldEntry := oldEntries[i]
+		newEntry := newEntries[i]
+
+		if !oldEntry.Equal(newEntry) {
+			return true
+		}
+	}
+	return false
+}
+
+func printArpEntries(log *base.LogObject, entries []arpEntry) {
+	for _, e := range entries {
+		log.Errorf("ARP entry - Out interface: %s, mac: %s, addr: %s, subnet: %s",
+			e.outIntf, e.macAddr, e.addr.String(), e.subnet.String())
+	}
+}
+
+func arpCmd(log *base.LogObject, ifname string, add bool, args ...string) (string, error) {
+	var out []byte
+	var err error
+
+	cmd := "arp"
+	cmdArgs := []string{"-i", ifname, "-d"}
+	if add {
+		cmdArgs[2] = "-s"
+	}
+	cmdArgs = append(cmdArgs, args...)
+	if log != nil {
+		log.Functionf("Calling command %s %v\n", cmd, args)
+		out, err = base.Exec(log, cmd, cmdArgs...).CombinedOutput()
+	} else {
+		out, err = base.Exec(log, cmd, cmdArgs...).Output()
+	}
+	if err != nil {
+		errStr := fmt.Sprintf("arp command %s failed %s output %s",
+			args, err, out)
+		if log != nil {
+			log.Errorln(errStr)
+		}
+		return "", errors.New(errStr)
+	}
+	return string(out), nil
+}
+
+// UpdateStaticArpEntries - Update static ARP entries between interfaces in same subnet
+func UpdateStaticArpEntries(ctx *DeviceNetworkContext, status types.DeviceNetworkStatus) {
+	log := ctx.Log
+	newArpEntries := makeArpEntries(log, makeArpGroups(status))
+	item, err := ctx.PubDeviceNetworkStatus.Get("global")
+	if err != nil {
+		// This might be the first DeviceNetworkStatus that we are publishing
+		// Add new ARP entries
+		printArpEntries(log, newArpEntries)
+		for _, e := range newArpEntries {
+			_, err := arpCmd(log, e.outIntf, true, []string{e.addr.String(), e.macAddr}...)
+			if err != nil {
+				log.Errorf("UpdateStaticArpEntries: Programming ARP failed with %s", err)
+			}
+		}
+		return
+	}
+	oldDNS := item.(types.DeviceNetworkStatus)
+
+	oldArpEntries := makeArpEntries(log, makeArpGroups(oldDNS))
+	if !arpEntriesDifferent(oldArpEntries, newArpEntries) {
+		log.Functionf("UpdateStaticArpEntries: No change in ARP entries")
+		return
+	}
+
+	// Delete old ARP entries
+	for _, e := range oldArpEntries {
+		_, err := arpCmd(log, e.outIntf, false, []string{e.addr.String()}...)
+		if err != nil {
+			log.Errorf("UpdateStaticArpEntries: Deleting ARP failed with %s", err)
+		}
+	}
+	// Add new ARP entries
+	printArpEntries(log, newArpEntries)
+	for _, e := range newArpEntries {
+		_, err := arpCmd(log, e.outIntf, true, []string{e.addr.String(), e.macAddr}...)
+		if err != nil {
+			log.Errorf("UpdateStaticArpEntries: Programming ARP failed with %s", err)
```

Comment:
Ditto

Reply:
will include the contents of entry in log